src/peft/tuners/tuners_utils.py

```diff
@@ -496,6 +522,7 @@ def inject_adapter(
                 unmatched_modules.append(key)
             else:
                 self.targeted_module_names.append(key)
+                self._check_target_module_compatiblity(peft_config, model, key)
```

Comment:
I missed this in the last review. For this to properly work, move this one line down by one line pass `target_name` instead of `key`, as `key` is the fully qualified name, like `'backbone.layers.0.mixer.conv1d'`.